### ISSUE TITLE
predict_ice return more outputs than (number of treatments * number of targets * 2 (T0, T1))

### ISSUE THREAD

**Issue by CoteDave** [Ibotta/mr_uplift#53]
Hi,

When I fit MRUplift() with 1 target and 2 treatments, predict_ice return and array with 4 columns. my interpretation is this
- col1: ice when T1=0
- col2:ice when T1=1
- col3:ice when T2=0
- col4:ice when T2=1
((number of treatments * number of targets * 2 (T0, T1)) = (2 * 1 * 2) = 4

When I fit MRUplift() with 1 target and 3 treatments, predict_ice return and array with 8 columns ????? I Don't understand why 8 ? 
should be 6 ?((number of treatments * number of targets * 2 (T0, T1)) = (3 * 1 * 2) = 6
Is this a bug ?

If not, How can we interpret the array returned from predict_ice as there is no column names ? 

![image](https://user-images.githubusercontent.com/49559881/196050599-11daf89f-7a4c-4b82-afd8-fabd5a21e397.png)

Thanks !

**Comment by CoteDave** [comment]
I found why 8 columns are outputed for 1 target with 3 treatments when .predict_ice

This line of code: 
![image](https://user-images.githubusercontent.com/49559881/196174301-d5982c59-c7ce-48ca-a3fb-a4038ded0774.png)

it seems to produce a ice_score for each combination of treatments and in this case, there are 8 combinations posssible. Soo we can't have a ice score for each binary response of each treatment... it's calculated as a whole... which makes sense !

but it's still difficult to know which treatment combination is associated with which score when you predict_ice... you need to dig to understand what's happening here.

**Comment by samcarlos** [comment]
Sorry I missed this …

Yup you got it! 

It’d be nice to have column names but 3-d np.arrays are messy enough to work with!

Also fwiw I generally look at the ERUPT metrics to validate the model and not the ICE output itself. The model can overfit pretty easily (or might not be calibrated) so I tend to look at the decisions the model makes more than the predicted responses. This is especially true when using the optized loss function.
